src/libraries/System.Private.CoreLib/src/System/DateTimeOffset.Android.cs

```diff
@@ -0,0 +1,97 @@
+        //
+        // On Android, loading AndroidTZData is expensive for startup performance.
+        // The fast result relies on `System.TimeZoneInfo.LocalDateTimeOffset` being set
+        // in the App Context's properties as the appropriate local date time offset from UTC.
+        // monovm_initialize(_preparsed) can be leveraged to do so.
+        // However, to handle timezone changes during the app lifetime, AndroidTZData needs to be loaded.
+        // So, on first call, we return the fast path and start a background thread to load
+        // the TimeZoneInfo Local cache implementation which loads AndroidTZData.
+        public static DateTimeOffset Now
+        {
+            get
+            {
+                DateTime utcDateTime = DateTime.UtcNow;
+
+                if (s_androidTZDataLoaded) // The background thread finished, the cache is loaded.
+                    return ToLocalTime(utcDateTime, true);
+
+                if (s_startNewBackgroundThread) // The cache isn't loaded and no background thread has been created
+                {
+                    lock (s_localUtcOffsetLock)
+                    {
+                        // Now may be called multiple times before a cache is loaded and a background thread is running,
+                        // once the lock is available, check for a cache and background thread.
+                        if (s_androidTZDataLoaded)
+                            return ToLocalTime(utcDateTime, true);
+
+                        if (s_loadAndroidTZData == null)
+                        {
+                            s_loadAndroidTZData = new Thread(() => {
+                                // Delay the background thread to avoid impacting startup, if it still coincides after 1s, startup is already perceived as slow
+                                Thread.Sleep(1000);
+
+                                _ = TimeZoneInfo.Local; // Load AndroidTZData
+                                s_androidTZDataLoaded = true;
+
+                                lock (s_localUtcOffsetLock)
+                                {
+                                    s_loadAndroidTZData = null; // Ensure thread is cleared when cache is loaded
+                                }
+                            }) { IsBackground = true };
+                        }
+                    }
+
+                    if (s_startNewBackgroundThread)
+                    {
+                        // Because Start does not block the calling thread,
+                        // setting the boolean flag to false immediately after should
+                        // prevent two calls to DateTimeOffset.Now in quick succession
+                        // from both reaching here.
+                        //
+                        // In the event multiple threads hit Start at the same time,
+                        // swallow the exception and move on.
+                        try
```

Comment:
Will someone create a PR to main to sync the code with the new changes made here?

Reply:
Yes